NGitLab.Mock/Pipeline.cs

```diff
@@ -12,7 +12,9 @@ public Pipeline(string @ref)
             Ref = @ref;
         }
 
-        public new Project Parent => (Project)base.Parent;
+        public Project Project => Parent is Pipeline pipeline ? pipeline.Project : (Project)Parent;
+
+        public bool IsParentAPipeline { get; set; }
```

Comment:
I'm not too convinced by `IsParentAPipeline`, it looks like a hack could you keep internal at least?